File: Dec2020/19Dec/ShuffleTheArray.java
/*
Problem Name: 1470. Shuffle the Array
Problem Link: https://leetcode.com/problems/shuffle-the-array/
*/
class Solution {
    public int[] shuffle(int[] nums, int n) {
        int x[] = new int[n], y[] = new int[n];
        for(int i=0;i<n;i++){
            x[i] = nums[i];
            y[i] = nums[i + n];
        }
        for(int i=0;i<n;i++){
            nums[i*2] = x[i];
            nums[i*2 + 1] = y[i];
        }
        return nums;
    }
}

File: Dec2020/19Dec/ShuffleTheArray.cpp
/*
Problem name: 1470. Shuffle the Array
Problem link: https://leetcode.com/problems/shuffle-the-array/
*/

class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> ans;
        bool flag=false;
        for(int i=0,j=n;j<2*n;){
            if(!flag){
                ans.push_back(nums[i]);
                i++;
                flag=true;
            }
            else{
                ans.push_back(nums[j]);
                j++;
                flag=false;
            }
        }
        return ans;
    }
};
